pkg/bpf/map_linux.go

```diff
@@ -158,14 +158,17 @@ func OpenMap(name string) (*Map, error) {
 
 	info, err := GetMapInfo(os.Getpid(), fd)
 	if err != nil {
+		unix.Close(fd)
 		return nil, err
 	}
 
 	if info.MapType == 0 {
+		unix.Close(fd)
 		return nil, fmt.Errorf("Unable to determine map type")
 	}
 
 	if info.KeySize == 0 {
+		unix.Close(fd)
```

Comment:
I was wondering if there was a reason why we can't use `defer(unix.Close(fd))` here after `ObjGet(name)`?

Reply:
aaaah yeah sorry indeed!